frontend/src/app/util/snap_data.cljs

```diff
@@ -79,33 +80,36 @@
                                           :pt %)))
         grid-x-data (get-grids-snap-points frame :x)
         grid-y-data (get-grids-snap-points frame :y)]
-    (-> page-data
-        ;; Update root frame information
-        (assoc-in [uuid/zero :objects-data frame-id] frame-data)
-        (update-in [parent-id :x] (make-insert-tree-data frame-data :x))
-        (update-in [parent-id :y] (make-insert-tree-data frame-data :y))
-
-        ;; Update frame information
-        (assoc-in  [frame-id :objects-data frame-id] (d/concat-vec frame-data grid-x-data grid-y-data))
-        (update-in [frame-id :x] #(or % (rt/make-tree)))
-        (update-in [frame-id :y] #(or % (rt/make-tree)))
-        (update-in [frame-id :x] (make-insert-tree-data (d/concat-vec frame-data grid-x-data) :x))
-        (update-in [frame-id :y] (make-insert-tree-data (d/concat-vec frame-data grid-y-data) :y)))))
+
+    (cond-> page-data
+      (not (ctl/layout-child? objects frame))
+
+      (-> ;; Update root frame information
+       (assoc-in [uuid/zero :objects-data frame-id] frame-data)
+       (update-in [parent-id :x] (make-insert-tree-data frame-data :x))
+       (update-in [parent-id :y] (make-insert-tree-data frame-data :y))
```

Comment:
Just a suggestion: prevent duplicated calls to `update-in` with common parent (in this case i'm refering to the `parent-id`), and replace it with `(update parent-id (fn [parent] (assoc ...)))`

Repeated calls to `update-in` creates the vector instance on each invocation.